### PR TITLE
Disallow validation on projects in draft or archive status for mappers

### LINKED ISSUE #919: Any user can validate on non-published projects
If a regular user visits a project that is archived, for example, they can validate/invalidate tiles.

Needs a check here: https://github.com/hotosm/tasking-manager/blob/develop/server/services/project_service.py#L136 like there is on https://github.com/hotosm/tasking-manager/blob/develop/server/services/project_service.py#L92-L93

### PR DESCRIPTION
**Pull request by ethan-nelson**
Originally, anyone was able to validate or invalidate tasks on a project whether it was in draft, archive, or published mode. This restricts validation in archived or draft statuses to only project managers.

Fixes #919. Part of #866 is also resolved with this (only the validating).

### CODE REVIEW

File: server/models/postgis/statuses.py [hunk 1]
```diff
@@ -57,6 +57,7 @@ class ValidatingNotAllowed(Enum):
     USER_NOT_VALIDATOR = 100
     USER_NOT_ACCEPTED_LICENSE = 101
     USER_NOT_ON_ALLOWED_LIST = 102
+    PROJECT_NOT_PUBLISHED = 103
```
Comment (ethan-nelson) [inline server/models/postgis/statuses.py]: These are defined in the postgis directory...does that mean we will need to add a database upgrade?

Comment (pgiraud) [reply server/models/postgis/statuses.py]: I'd be interested to know as well. I thought it was an ORM only thing but it looks like postgreSQL uses theses value internally. Most probably to add a constraint on the possible values.
What I don't get though is that we never use this class for any table in the model. Looks like there's no need for an Enum here.
Oh wait! This is not an `SQLAlchemy Enum`. It's from the python core library.
No database migration required, IMHO.

**Comment by ethan-nelson** [comment]
The most recent update ensures that project managers can still validate on projects that are not published (i.e. archived or draft).

**Comment by russdeffner** [comment]
@ethan-nelson why just validating? I like to test my projects in draft (i.e. map) and often  we do splitting, etc. - could see pm wanting to mark the bad imagery themselves as well before publishing. Is there another issue for 'all functions for pm's in draft/archive'?

**Comment by ethan-nelson** [comment]
Hey @russdeffner, thanks for the feedback. Yes that functionality for mapping is definitely on the list to do and is possible following similar to what I did here--I may do it here in a few minutes. I just didn't want to combine the validation and mapping additions in the same PR. Hope you have a good holiday!

**Pull request merged**